scripts/generate-renode-scripts.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+import csv
+import zlib
+import argparse
+
+def generate_ethmac(data):
+    base = get_entry_value(data, 'csr_base', 'ethmac')
+    buffer_base = get_entry_value(data, 'memory_region', 'ethmac')
+    buffer_size = get_entry_value(data, 'memory_region', 'ethmac', 1)
+    interrupt = get_entry_value(data, 'constant', 'ethmac_interrupt')
+
+    result = [
+        'ethmac: Network.LiteX_Ethernet @ {',
+        '       sysbus <{}, +0x100>;'.format(base),
+        '       sysbus new Bus.BusMultiRegistration {{ address: {}; size: {}; region: "buffer" }}'.format(buffer_base, buffer_size),
+        '    }'
+    ]
+
+    if interrupt is not None:
+        result.append('    -> cpu@{}'.format(interrupt))
+    return result
+
+def generate_peripheral(data, name, model, properties=None):
+    base = get_entry_value(data, 'csr_base', name)
+    interrupt = get_entry_value(data, 'constant', '{}_interrupt'.format(name))
+
+    result = ['{}: {} @ sysbus {}'.format(name, model, base)]
+    if properties is not None:
+        for pname, pvalue in properties.items():
+            if pvalue is not None:
+                result.append('    {}: {}'.format(pname, pvalue))
+    if interrupt is not None:
+        result.append('    -> cpu@{}'.format(interrupt))
+    return result
+
+def generate_memory_region(data, name):
+    base = get_entry_value(data, 'memory_region', name)
+    size = get_entry_value(data, 'memory_region', name, 1)
+
+    return [
+        '{}: Memory.MappedMemory @ sysbus {}'.format(name, base),
+        '    size: {}'.format(size)
+    ]
+
+def generate_silencer(data, name):
+    base = get_entry_value(data, 'csr_base', name)
+
+    return [
+        'sysbus:',
+        '    init add:',
+        '        SilenceRange <{} 0x200> # {}'.format(base, name)
+    ]
+
+def generate_flash_boot(data):
+    # TODO: this value is hardcoded, as I couldn't find it in the csv
+    base = 0x20228000
+    return [
+        'boot_flash: Memory.ArrayMemory @ sysbus {}'.format(base),
+        '    size: 0x00100000'
+        ]
+
+def generate_cpu(data):
+    kind = get_entry_value(data, 'constant', 'config_cpu_type')
+
+    if kind == 'VEXRISCV':
+        return ['cpu: CPU.VexRiscv @ sysbus']
+    elif kind == 'PICORV32':
+        return [
+            'cpu: CPU.PicoRV32 @ sysbus',
+            '    cpuType: "rv32imc"'
+        ]
+    else:
+        raise Exception('Unsupported cpu type: {}'.format(kind))
+
+def get_entry_value(data, kind, name, idx=0):
+    for entry in data:
+        if entry[0] == kind and entry[1] == name:
+            return entry[idx + 2]
+    return None
+
+def generate_repl(data):
+    return \
+        generate_cpu(data) + \
+        generate_memory_region(data, 'rom') + \
+        generate_memory_region(data, 'sram') + \
```

Comment:
You should loop over the memory regions define in either `mem.h` or `csr.csv`;
```python

data = """
memory_region,rom,0x00000000,32768,
memory_region,sram,0x10000000,32768,
memory_region,spiflash,0xa0000000,16777216,
memory_region,main_ram,0x40000000,268435456,
memory_region,ethmac,0xb0000000,8192,
"""
for line in csv.reader(data):
  if line[0] != "memory_region":
    _, memory_name, memory_start_adress, memory_size = *line
   # Some peripherals (like Ethernet) define a memory region, as peripherals will already be defined and taken ownership of the region we skip creating it here.
   if model.already_has_memory(memory_name):
      print("Skipping {} as already claimed by peripheral {}".format(memory_name, model.already_has_memory(memory_name))
      continue
    generate_memory_region(memory_name, memory_start_address, memory_size)
```

Reply:
I created a separate `parse_csv` function that does the job.